src/uniqueElements.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @since 0.0.7
+ */
+import { isDeepEq } from "./Is/deepEq";
+
+type Duplicate<T> = {
+  indexs: number[];
+  content: T;
+};
+
+/**
+ * filter unique elements of an array
+ * and locate the duplication indexs and it's content
+ *
+ * @since 0.0.7
+ */
+export const uniqueElements = <T>(a: T[]) => {
+  const uniqueElements: T[] = [];
+  const duplicates: Duplicate<T>[] = [];
+  const cache = new Set<number>();
+
+  for (let i = 0; i < a.length; i++) {
+    if (cache.has(i)) {
+      continue;
+    }
+
+    const current = a[i];
+    const duplicateIndexs = [];
+    for (let j = i + 1; j < a.length; j++) {
```

Comment:
it's o(n<sup>2</sup>) since it has a loop inside of loop with quadratic linear iteration #cmiiw
we had many approaches to solving this problem, how about we try a performance test to compare each of the solutions

Reply:
indeed it requires 2 loop, but it has circuit breaker to exit the loop if the item already. no Idea how to solve this with one loop haha

Reply:
😄 since all of the solutions might have similar o notation, let's try to compare the performance when we get a chance